### PR TITLE
Attempt to provide metrics for vertx bytebuf allocator

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

**Comment by vietj** [comment]
ping @franz1981

File: src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java [hunk 1]
```diff
@@ -53,4 +50,17 @@ public boolean release() {
   public boolean release(int decrement) {
```
Comment (franz1981) [inline src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: Given that the logic for the reference count seems to be disabled here (see `::release` code pointed here)
how do we know that `freeArray` will be correctly fired?
I haven't downloaded this branch locally, but I'm not sure this is going to work

Comment (vietj) [reply src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: I did like what is in `UnpooledByteBufAllocator.InstrumentedUnpooledHeapByteBuf` class

Comment (franz1981) [reply src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: yeah but our implementations doesn't make free to be called, try with a debugger (maybe I'm wrong, but it looks like that)

Comment (vietj) [reply src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: you are right, I think we should handle this through finalize/cleaner instead ?

Comment (vietj) [reply src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: or maybe don't account for these buffers ?

Comment (franz1981) [reply src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: I prefer this option, unless we are ready to pay the price for reference counting and "activate" free for such buffers as well

Comment (vietj) [reply src/main/java/io/vertx/core/buffer/impl/VertxHeapByteBuf.java:50]: I do agree with you, so this PR is actually not needed at all.

**Pull request closed without merging**